VocaDb.ResXFileCodeGenerator/SourceGenerator.cs

```diff
@@ -1,146 +1,41 @@
-		var publicClassGlobal = false;
-		if (context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.ResXFileCodeGenerator_PublicClass", out var publicClassSwitch))
-			publicClassGlobal = publicClassSwitch.Equals("true", StringComparison.OrdinalIgnoreCase);
-
-		var nullForgivingOperators =
-			context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.ResXFileCodeGenerator_NullForgivingOperators", out var nullForgivingOperatorsSwitch) &&
-			nullForgivingOperatorsSwitch is { Length: > 0 } &&
-			nullForgivingOperatorsSwitch.Equals("true", StringComparison.OrdinalIgnoreCase);
-		
-		var staticClass = !(context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.ResXFileCodeGenerator_StaticClass", out var staticClassSwitch) &&
-			staticClassSwitch is { Length: > 0 } &&
-			staticClassSwitch.Equals("false", StringComparison.OrdinalIgnoreCase));
-
-		var resxFiles = context.AdditionalFiles
-			.Where(af => af.Path.EndsWith(".resx"))
-			.Where(af => Path.GetFileNameWithoutExtension(af.Path) == GetBaseName(af.Path));
-
-		foreach (var resxFile in resxFiles)
+		context.RegisterSourceOutput(inputs, (ctx, file) =>
 		{
-			using var resxStream = File.OpenRead(resxFile.Path);
-
-			var options = new GeneratorOptions(
-				LocalNamespace:
-					GetLocalNamespace(
-						resxFile.Path,
-						context.AnalyzerConfigOptions.GetOptions(resxFile).TryGetValue("build_metadata.EmbeddedResource.TargetPath", out var targetPath) && targetPath is { Length: > 0 }
-							? targetPath
-							: null,
-						projectFullPath,
-						rootNamespace),
-				CustomToolNamespace:
-					context.AnalyzerConfigOptions.GetOptions(resxFile).TryGetValue("build_metadata.EmbeddedResource.CustomToolNamespace", out var customToolNamespace) && customToolNamespace is { Length: > 0 }
-						? customToolNamespace
-						: null,
-				ClassName: Path.GetFileNameWithoutExtension(resxFile.Path),
-				PublicClass:
-					context.AnalyzerConfigOptions.GetOptions(resxFile).TryGetValue("build_metadata.EmbeddedResource.PublicClass", out var perFilePublicClassSwitch) && perFilePublicClassSwitch is { Length: > 0 }
-						? perFilePublicClassSwitch.Equals("true", StringComparison.OrdinalIgnoreCase)
-						: publicClassGlobal,
-				NullForgivingOperators: nullForgivingOperators,
-				StaticClass:
-					context.AnalyzerConfigOptions.GetOptions(resxFile).TryGetValue("build_metadata.EmbeddedResource.StaticClass", out var perFileStaticClassSwitch) && perFileStaticClassSwitch is { Length: > 0 }
-						? !perFileStaticClassSwitch.Equals("false", StringComparison.OrdinalIgnoreCase)
-			 			: staticClass
-			);
-
-			var source = s_generator.Generate(resxStream, options);
-
-			context.AddSource($"{options.LocalNamespace}.{options.ClassName}.g.cs", source);
-		}
+			var filecontent = file.File.GetText(ctx.CancellationToken);
+			if (filecontent == null) return;
```

Comment:
Please replace `== null` with `is null`.

Reply:
done